### PR TITLE
 feat: add room-state tests

### PR DESCRIPTION
**Pull request by meenal06**
Signed-off-by: Meenal Trivedi <meenaltrivedi6102@gmail.com>

### CODE REVIEW

**Comment by meenal06** [comment]
@kegsay please review the PR!

**Comment by meenal06** [comment]
@kegsay ping

File: tests/apidoc_room_state_test.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	if err != nil {
+		t.Fatalf("unable to read response body: %v", err)
+	}
+
+	roomID = gjson.GetBytes(body, "room_id").Str
+	roomAlias = gjson.GetBytes(body, "room_alias").Str
+	return roomID, roomAlias
+}
+
+func generateRoomAlias() (string, error) {
```
Comment (kegsay) [inline tests/apidoc_room_state_test.go:99]: Why are we doing this? It seems a little silly to randomly generate an alias every time. It hurts reproducibility of tests as each test run now has different values, and there is no improvement to clarity using `generateRoomAlias()` over a hardcoded value of say `"test_alias"`. Removing this also means you don't need to check for `error`.

Instead, just add an argument to `roomFixture` which is `roomAliasName string` and use that.

File: tests/apidoc_room_state_test.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+func TestRoomState(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	authedClient := deployment.Client(t, "hs1", "@alice:hs1")
+	t.Run("Parallel", func(t *testing.T) {
+		// sytest: GET /rooms/:room_id/state/m.room.member/:user_id fetches my membership
+		t.Run("GET /rooms/:room_id/state/m.room.member/:user_id fetches my membership", func(t *testing.T) {
+			t.Parallel()
+			roomID, _ := roomFixture(t, authedClient)
```
Comment (kegsay) [inline tests/apidoc_room_state_test.go:26]: I don't understand why we need `roomFixture` when `roomID := authedClient.CreateRoom(t, map[string]interface{}{})` is not only clearer but doesn't need `_` on the alias.

File: tests/apidoc_room_state_test.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+					match.JSONKeyPresent("events_default"),
+					match.JSONKeyPresent("users"),
+					match.JSONKeyPresent("events"),
+				},
+			})
+		})
+	})
+}
+
+func roomFixture(t *testing.T, authedClient *client.CSAPI) (roomID string, roomAlias string) {
```
Comment (kegsay) [inline tests/apidoc_room_state_test.go:75]: I am not convinced we need this function at all, remove it please.

**Review by kegsay — CHANGES_REQUESTED** [review]

File: tests/apidoc_room_state_test.go (hunk 3 above)
Comment (meenal06) [reply tests/apidoc_room_state_test.go:75]: I didn't know the existence of CreateRoom until my last PR, apologies for this. I will clean up all unnecessary code

**Review by kegsay — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**